### ISSUE TITLE
Misleading example for per-sample gradient

### ISSUE THREAD

**Issue by mingfeisun** [pytorch/tutorials#2827]
In the example of per-sample gradient, the following line can be misleading since the `predictions` of a net are logits:  
https://github.com/pytorch/tutorials/blob/08a61b7cae9d00312d0029b1f86a248ec1253a83/intermediate_source/per_sample_grads.py#L49

The correct way should be: 
``` python
return F.nll_loss(F.log_softmax(predictions, dim=-1), targets) 
```

Would appreciate if this can be corrected.

**Comment by jmarintur** [comment]
Hi @mingfeisun, looking at the network used in the example, it does seem there's an error:

https://github.com/pytorch/tutorials/blob/08a61b7cae9d00312d0029b1f86a248ec1253a83/intermediate_source/per_sample_grads.py#L44-L46

the output is overwritten after applying the F.log_softmax in L45. If we remove that line, all should be good. Wdyt @albanD ?

**Comment by jmarintur** [comment]
Here's the [PR](https://github.com/pytorch/tutorials/pull/2839) @albanD.
